### PR TITLE
[#2138] Relicense to LGPLv3+/GPLv2+

### PR DESCRIPTION
**Pull request by SwooshyCueb**
In service of #2138

In order to absorb libs3 into the project, we must relicense to a compatible license with the existing libs3 code.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,12 +1,14 @@
-Copyright (c) 2014, The University of North Carolina at Chapel Hill
-All rights reserved.
+Copyright (c) 2023, The iRODS Consortium
```
Comment (trel) [inline LICENSE:1]: we don't exist as 'The iRODS Consortium'... just legally 'The University of North Carolina at Chapel Hill'

File: LICENSE [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 
-Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+The iRODS s3 Resource Plugin is free software: you can redistribute it and/or
```
Comment (trel) [inline LICENSE:3]: s3 -> S3

File: LICENSE [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+modify it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, version 3 or above of the License.
 
-- Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+The iRODS s3 Resource Plugin is distributed in the hope that it will be
```
Comment (trel) [inline LICENSE:7]: s3 -> S3

File: LICENSE [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-
-- Neither the name of the University of North Carolina at Chapel Hill nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
-
-THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+You should have received a copy of the GNU Lesser General Public License
+version 3 along with The iRODS s3 Resource Plugin, in a file named
```
Comment (trel) [inline LICENSE:13]: s3 -> S3

File: CMakeLists.txt [hunk 5]
```diff
@@ -180,7 +180,7 @@ set(CPACK_DEBIAN_PACKAGE_CONTROL_EXTRA "${CMAKE_SOURCE_DIR}/packaging/postinst;"
 set(CPACK_DEBIAN_FILE_NAME DEB-DEFAULT)
 
 set(CPACK_RPM_COMPONENT_INSTALL OFF)
-set(CPACK_RPM_PACKAGE_LICENSE "BSD-3-Clause")
+set(CPACK_RPM_PACKAGE_LICENSE "LGPL-3.0-or-later")
```
[the remark this answers is not in the document]
Comment (SwooshyCueb) [reply CMakeLists.txt:183]: This is the proper short identifier:
https://spdx.org/licenses/LGPL-3.0-or-later.html

[the remark this answers is not in the document]
Comment (trel) [reply CMakeLists.txt:183]: sold.

**Comment by trel** [comment]
i think we're there on the licensing front - now just need the actual code and confirmation nothing is broken.

**Comment by SwooshyCueb** [comment]
Should that be a separate PR or do you want me to append it to this one and rename it?

**Comment by SwooshyCueb** [comment]
After seeing some discussion on fedi, I've been doing a little bit more reading up on the GPL family of licenses. Added in the GPLv3 and LGPLv3 was a provision that opposes digital rights management. This provision scared away a lot of developers from adopting the licenses, and many groups have done their best to avoid software using these licenses.

Sources:
- https://lwn.net/Articles/202106/
- https://www.cnet.com/tech/tech-industry/torvalds-no-gpl-3-for-linux/
- https://meta.ath0.com/2012/02/05/apples-great-gpl-purge/
- https://archive.org/details/copyleftconf2020-allison

libs3 is licensed under LGPLv3+ *or* GPLv2+. Allowing distribution under the GPLv2 provides an out for vendors who don't want to deal with the apparently troublesome provision. I think we should do the same. I'll be updating this PR accordingly.

**Comment by trel** [comment]
that makes sense.  i'm familiar with those concerns as well.

**Comment by SwooshyCueb** [comment]
\#'d

**Pull request merged**